### ISSUE TITLE
safari-is-the-new-ie.com

### ISSUE THREAD

**Issue by stuartpb** [h5bp/lazyweb-requests#181]
I just bought [this domain](http://safari-is-the-new-ie.com), which perfectly encapsulates something I've been saying for that last year or two: Microsoft is no longer the burden on the web platform it once was, and Apple has risen up to take its frustrating place.

When I first registered it, I had the domain redirecting to [Nolan Lawson's terrific blog post by this name](http://nolanlawson.com/2015/06/30/safari-is-the-new-ie/), which I now link to as part of a GitHub Pages site:

https://github.com/stuartpb/www.safari-is-the-new-ie.com/

I'm trying to build a ~single-screen distillation of that article's points for the average user (in the vein of the sites that encouraged people to ditch IE like http://browsingbetter.com/ and http://abetterbrowser.org/), along with a "What Can I Do?" section making a few possible calls to action:
- Drumming up public support for telling Apple to get their act together
- Petitioning apple to [allow alternative iOS WebView implementations](https://twitter.com/jaffathecake/status/612992537238896641) if they're not going to maintain WebKit to the level of Blink/Gecko/Spartan
- Encouraging devs who are capable to try to bring WebKit up to Blink and co. themselves, and/or contribute to polyfills

Thoughts? Pinging authors of pages mentioned above / linked in the article: @nolanlawson @jakearchibald @simevidas @yathit @thatryan @ejc

Also pinging @burocratik (http://outdatedbrowser.com/en) and @xPaw (author of CloudFlare's A Better Browser banner-inliner https://github.com/xPaw/CF-ABetterBrowser)

**Comment by Garbee** [comment]
> start petitioning the FTC to ready an anti-trust suit against Apple

This would fundamentally fail. Apple is not in a market dominant position. They are *A* big player, but Android is just as large, if not larger. FTC would have no grounds to take on a lawsuit. Also, the MS situation was a bit different. They were using their market dominance to move into other markets and outright bully people into doing things that worked best for them. Apple has no such control over the web with Safari (nor does any browser.)

I do agree, Apple is holding things back now, but rattling anti-trust suits that have no merit other than "We don't like what they are doing" is a bit extreme.

**Comment by stuartpb** [comment]
Yeah, to be honest I kind of felt a little extremist to suggest it when I was writing the original post, but I included it anyway because I *do* personally feel that Apple is "using their market dominance to move into other markets and outright bully people into doing things that worked best for them." (How many apps in the Valley do you see *only* launching iOS versions of their apps because of this platform disparity? Nearly all of them - which Apple is using to their advantage to push into new markets with entries like the Apple Watch.)

Nonetheless, I'll take that part out for now so as to not pollute / distract from the overall message. For future reference, the line that was removed:

> - If Apple doesn't do either, start petitioning the FTC to ready an anti-trust suit against Apple for doing the same things Microsoft was doing in the nineties (except worse, because of their platform lockout)

**Comment by auchenberg** [comment]
I like the idea of drumming up the community to put pressure on Apple to allow other browser engines on iOS (both WebView and standalone). We could easily put a website together and start collecting signatures from industry leaders and companies. 

That would be the community-version of an anti-trust case :)

**Comment by yathit** [comment]
@auchenberg :+1:

**Comment by auchenberg** [comment]
Ideas for a campaign domain?

- allow-other-browers-on-ios.com
- save-the-web-on-ios.com
- save-the-mobile-web.com
- better-ios-web.com
- better-mobile-web.com
- better-web-on-ios.com
- move-the-web-forward.com
- set-the-web-free.com

**Comment by ejc** [comment]
far-too-many-dashes-dot-net.com

**Comment by stuartpb** [comment]
@ejc I used hyphens because `safariisthenewie` reads like a portmanteau you'd see if James Joyce wrote Dune fanfiction: "Safar 'i'is, then, Ewie?"

@auchenberg I think safari-is-the-new-ie.com/petition, with a forkable mechanism for signing (like extensiblewebmanifesto.org, but with GitHub instead of Google Forms) would be fine. Maybe we could get a few signatures in this thread to start us off?

**Comment by auchenberg** [comment]
@stuartpb Sure that could work. Let's get the ball rolling. However with Google Form is a bit easier for people sign up than having to edit/fork on GitHub.

**Comment by stuartpb** [comment]
@auchenberg Yeah, but I'm approaching this from the point of view that we primarily want developers in terms of the *names* that are on the list, and for devs it's easy enough to visit this issue and post a comment. (When it comes to collecting *end user* signatures, I'd rather cross that bridge when we come to it.)

**Comment by auchenberg** [comment]
@stuartpb Sure, but GitHub is still more clicks away than a simple Google Form. Anyways, let's get the ball rolling. Do you create a repo? Use extensiblewebmanifesto.com as boilerplate?

**Comment by stuartpb** [comment]
I've created a rough initial page from scratch: https://github.com/stuartpb/www.safari-is-the-new-ie.com

I welcome any pull requests to make this look a bit classier.

**Comment by jakearchibald** [comment]
Really like this idea & hope it'd put pressure in the right places. I think it'd weaken the project if people paid by other browser vendors worked on it though.

**Comment by stopsatgreen** [comment]
I like the idea - although I don’t agree that it’s ‘the new IE’, I’d like to see pressure put on them to release more frequently, and keep up with other browsers to give us a more consistent platform.

**Comment by jbrooksuk** [comment]
:+1: for all of this.

**Comment by mickaelandrieu** [comment]
Apple (MacOS X + iOS) is only 10% of the OS market, maybe 15%.

If you don't want to be locked by their politics, you have 2 choices:
* back to Windows or linux for desktop OS (and enjoy nice and updated web browsers)
* back to windows phone (bad idea) or android, and also enjoy nice and updated mobile web browser

Apple is not a problem, because you don't have to buy it.

**Comment by glivera** [comment]
!!!!!!!

**Comment by jpsc** [comment]
@mickaelandrieu if you are being sarcastic you are doing a excellent job. Since when is this a problem for us developers and tech-savvy people? This to improve life of everybody that has iOS/OSX devices. Those people you can't make them change those things. Even if you could, that would be like chopping a head when you have a headache.

**Comment by Garbee** [comment]
> Apple is not a problem, because you don't have to buy it.

But, developers do need to support their platform since it is widely used. So long as they are slow rolling out new technology (stuck on an annual release for big additions it seems) then they are holding the web back.

Simply not being super-dominant market-wise is not the only case in play. They are super-dominant for some sites and applications that means they need to be supported. Along with pretty much any e-commerce site since it would suck to turn 10-15% of the total market away simply because they chose Apple software which makes the choice for them on what to use and they don't know any better.

Safari is not evergreen, this is a problem.

**Comment by mickaelandrieu** [comment]
It's not about sarcatism: if we start to make support like we have done before for IE, things won't change.

Apple want customers, it's not a development problem **but a platform** problem.

A lot of companies don't provide - for instance - windows phone applications of IE <8 support, I don't see "why" for Apple products developpers have to work hard to make things work as expected.

As @Garbee said,

> Safari is not evergreen, this is a problem.

As Safari is not open source, and as Apple refuse to let others real web browsers to be installed in Apple products:

> If Apple doesn't do either, start petitioning the FTC to ready an anti-trust suit against Apple for doing the same things Microsoft was doing in the nineties (except worse, because of their platform lockout)

:+1: this make sense.

**Comment by jpsc** [comment]
I understand you point, but if we can make it noise and try to make them change. I know it's probably talking to a big white wall, but still.

And its not about developers working harder; If the technology is not there, you just don't use it. Progressive Enhancement like on the blog post. 

In the end is your product that suffers with it, because the user is not going to blame Apple for a website/webapp, they are going to blame the product itself. 

So its pretty clear what Apple wants with this. Give a not so go experience on the web so the users/stakeholders go for a native app.

**Comment by nolanlawson** [comment]
I don't know if this campaign is the right approach.

My article was kinda harsh, and this site is a good way to continue shaming Apple, but at some point I want to stop bashing them and let the healing begin. I'd like to see Apple employees come to meetups and conferences. I want them to participate more in these new APIs. I want them to push the web forward the way they were doing 5 years ago.

So I worry that by continuing to heap on the scorn, we're just going to make them dig their heels and get further estranged from the web community. I'm an outsider, so it's okay for me to write stuff that loses me a few friends at Apple. But now that the damage has been done, maybe we should take this as an opportunity to extend a hand of friendship to Apple.

The site itself is great, and I am perfectly okay with the content. But as for the petition, maybe it could have some more constructive demands:

* Allow other rendering engines, big yes to that.
* More outreach and contact with the community &ndash; where is Apple's Jake Archibald? Where is their Christian Heilmann? Where is the person who's going out and giving talks and getting developers excited about Safari?
* Open up the Radar bugtracker. WebKit bugtracker is nice, but many of the issues end up getting funneled into the private tracker.
* Or, create a forum where people can vote on issues they care about, ala Microsoft's [uservoice.com](https://wpdev.uservoice.com/forums/257854-microsoft-edge-developer).
* Other ideas?

Not trying to totally diss on your idea, and it's definitely my fault for writing the article in the first place, but just thinking maybe a more positive tone would help. :smiley:

**Comment by grayghostvisuals** [comment]
@nolanlawson Well said. No need for public shaming through a site. What happens when another browser craps on us? Make another website? Not the best way towards a solution here.

A better approach would be to encourage as mentioned in the above comment and show examples of where they are failing compared to the rest. The pudding will prove this just give Apple time.

**Comment by egojab** [comment]
Where is the site to shame devs from "only works in Chrome" stuff? What's worse, a browser vendor not supporting bleeding edge stuff, or a bunch of developers only supporting one browser?

Let's be honest, when it comes to being the "next IE" sites that only work in Chrome are far worse than Safari not supporting bleeding edge stuff. Safari, like every browser, certainly has it's problems, but it's made for the web _now_, not the future web. I have far less a problem with that than with any site that only works in any specific browser, and there are far too many Chrome-only sites.

**Comment by stuartpb** [comment]
@nolanlawson 

> So I worry that by continuing to heap on the scorn, we're just going to make them dig their heels and get further estranged from the web community. I'm an outsider, so it's okay for me to write stuff that loses me a few friends at Apple. But now that the damage has been done, maybe we should take this as an opportunity to extend a hand of friendship to Apple.

Completely agree - if we want anything to get better, the site's tone has to convey a spirit of hopeful reconciliation, with as much de-emphasis as possible on the scorn and shame. It's important that we're clear about what exactly we're petitioning for early on.

Where I would put it is that it's a petition for Apple to open up their development by taking a few small, clear, reasonable measures that will let us take steps toward fixing WebKit's problems *without* requiring major work on Apple's part. Building off the ones you mentioned:

- Remove [guideline 2.17](https://developer.apple.com/app-store/review/guidelines/#functionality) "Apps that browse the web must use the iOS WebKit framework and WebKit Javascript" from the Apple App Store Review Guidelines, so that developers are allowed to implement their own solutions. (As Android has shown, devs usually only use this measure if they *really* have to.)
- Start posting discussion *inside* the WebKit bug tracker and/or another open bug tracker for Safari, rather than linking to internal Radar discussions that aren't open to the public. This would let open source contributors and other members of the web platform community to jump in and proffer solutions, or even fixes, as well as discuss the standards themselves.

While I do want to see Apple participating in the standards space again, and to have community outreach people, cultural shifts like this are big ideas, which would require large-scale changes on Apple's part. I'd rather have the petition focus on the tipping point where *small* changes on Apple's part will make a *large* difference to the community.

@grayghostvisuals Showing specific examples of where they are failing is why I'm linking to Nolan's article early, though I'm thinking I might flip it around and put the "Sign the petition" C2A early and have "More information" beneath that, patterning after the way pages like Fight for the Future's campaigns are laid out.

**Comment by brucelawson** [comment]
.. and I was going to suggest http://safarie6.com/ as a domain name.

**Comment by stuartpb** [comment]
@brucelawson Just registered it as a redirect - I was originally going to pick "safarie" as a domain name, but some nasty malware bundler is sitting on it (and I didn't want to specify "IE6" in the URL because IE8 has the new IE6 for years - Safari is more like the new IE8).

**Comment by scshepard** [comment]
hi, enjoying the dialog here. it is my understanding that mac unix allows other browsers, correct or wrong?

**Comment by stuartpb** [comment]
@scshepard The full desktop OSX for MacBooks and the like allows other browsers, but iOS (as used on iPhones and iPads) does not: [due to a restriction Apple places on apps it allows on the app store](https://developer.apple.com/app-store/review/guidelines/#functionality), any browser on iOS has to effectively wrap Safari.

This presents a major obstacle for the mobile web, as this restriction, combined with Apple's reticence in introducing new features / fixes to Safari, leaves mobile web devs forced to work within years-old constraints if they want to support iOS.

This is why removing this restriction from Apple's App Store Review Guidelines is [the first item on the petition](https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-117754106) - removing this restriction on iOS would allow users to move the web forward themselves by using other browsers, the same way as they did on the PC when Internet Explorer started getting surpassed by the rest of the web.

**Comment by simevidas** [comment]
I'd like to make a quick point that there's no need to rush, as two important events are yet to happen:

1. the Safari 9 release,
2. Microsoft demonstrating that Edge is evergreen (specifically, web platform features being shipped via automatic updates, in relatively short intervals)

There’s going to be an increasing amount of a activity around this issue, which is great, but I wouldn’t want the “main event” to happen before November or December at the earliest.

**Comment by stuartpb** [comment]
@jakearchibald:

> Really like this idea & hope it'd put pressure in the right places. I think it'd weaken the project if people paid by other browser vendors worked on it though.

I regrettably agree - the conflict of interest (to put it mildly) would muddy the waters, were support for this campaign come from Safari's direct competitors.

That said, I think it *would* be powerful, without sending a mixed message, if major-destination Web players *without* skin in the browser game, and other grassroots-campaign organizations, were to voice their support. I'm thinking of people from houses like:

- Facebook
- EFF
- GitHub
- Wikimedia
- Reddit
- Stack Overflow
- Adobe

Thoughts? Do any developers from these sites have experiences with their struggles in supporting iOS?

**Comment by stuartpb** [comment]
I'm trying to link to a straightforward visual breakdown of where Safari falls short relative to the competition - I pulled up [this table](http://caniuse.com/#compare=safari+9,ios_saf+9,chrome+43,and_chr+42,firefox+38,and_ff+38,ie+Edge) on caniuse, but it doesn't really visually demonstrate the problem, since old standards (like Web SQL, which has been deprecated for years), trivial details (like multiple file selection, which is only unsupported by Android's file selection UI), and insignificant features (when's the last time you even *heard* somebody mention JPEG 2000?) where Safari still has a green bar have the same weight as significant, unpolyfillable modern standards like getUserMedia, Beacon, User Timing, and WeakMap/WeakSet (which, as an ES6 feature, isn't even on this table).

**Comment by nolanlawson** [comment]
@stuartpb Since the problem is that certain features are more important/desirable than others, could the solution be a forum system where people could vote up one feature or another? For instance, I might vote for things like WEBM, WEBP, WebRTC, fixing IndexedDB, Service Worker, Web Animations, Web Manifest, and `canvas.toBlob()`. But everybody has their own wishlist. :)

**Comment by stuartpb** [comment]
@nolanlawson That sounds like a job for an open issue tracker (ie. that's what Chrome has, issues that you can "star" to upvote).

**Comment by simevidas** [comment]
@nolanlawson I was going to suggest an unofficial UserVoice forum (analogous to what Microsoft is [having for Edge](https://wpdev.uservoice.com/forums/257854-project-spartan-platform)), but it’s very pricey.

**Comment by stuartpb** [comment]
@simevidas @nolanlawson I was going to suggest something like https://github.com/isaacs/github/issues for Safari, but we already do have http://openradar.appspot.com/ - maybe just add starring to that app?

**Comment by stuartpb** [comment]
And of course [the WebKit bug tracker is still open][wkbug] - if we could provide some kind of drop-in replacement for Bugzilla that wasn't quite so... CGI, maybe that would help things be a bit less of a mess on the issue tracker front.

[wkbug]: https://bugs.webkit.org/buglist.cgi?component=Accessibility&component=ANGLE&component=Animations&component=Bindings&component=Canvas&component=CSS&component=Evangelism&component=Event%20Handling&component=Forms&component=Frames&component=History&component=HTML%20DOM&component=HTML%20Editing&component=HTML%20Events&component=Images&component=Java&component=JavaScriptCore&component=Layout%20and%20Rendering&component=MathML&component=Media%20Elements&component=New%20Bugs&component=Page%20Loading&component=PDF&component=Platform&component=Plug-ins&component=Printing&component=SVG&component=Tables&component=Text&component=Tools%20%2F%20Tests&component=Web%20Audio&component=WebCore%20JavaScript&component=WebCore%20Misc.&component=WebGL&component=Web%20Inspector&component=Web%20Inspector%20%28Deprecated%29&component=WebKit2&component=WebKit%20API&component=WebKit%20BlackBerry&component=WebKit%20BREWMP&component=WebKit%20EFL&component=WebKit%20Gtk&component=WebKit%20Misc.&component=WebKit%20Qt&component=WebKit%20Website&component=WebKit%20wx&component=Web%20Template%20Framework&component=XML&component=XML%20DOM&product=WebKit&query_format=advanced&order=changeddate%20DESC%2Cbug_status%2Cpriority%2Cassigned_to%2Cbug_id&query_based_on=

**Comment by addyosmani** [comment]
@auchenberg https://kenneth.io/blog/2015/07/03/safari-isnt-the-problem-but-the-lack-of-browser-choice-in-ios-is/ resonated with me. It's an excellent write-up.

> More outreach and contact with the community – where is Apple's Jake Archibald? Where is their Christian Heilmann? Where is the person who's going out and giving talks and getting developers excited about Safari?

That would be Jonathan Davis: https://twitter.com/jonathandavis He's an evangelist for Apple with a focus on Safari. He recently joined, but there's value in trying to collaborate with him to get the message here heard by the right folks. We might also consider reaching out to Kevin Decker who is the Safari engineering manager and may also be interested (https://twitter.com/DeckerKev) or Edward O'Connor (https://twitter.com/hober) who works on the web standards work for WebKit/Safari at Apple.

> Or, create a forum where people can vote on issues they care about, ala Microsoft's uservoice.com.

UserVoice was pretty successful for MS, but afaik was driven directly by the IE team. If something ends up getting crafted here, we'll probably want it done in a way that there's genuine consideration paid to suggestions and it doesn't just become a dumping ground for 'please implement X' if Apple are ignoring it.

> I think it would be powerful, without sending a mixed message, if major-destination Web players without skin in the browser game, and other grassroots-campaign organizations, were to voice their support. I'm thinking of people from houses like:

There's a decent chance mobile web experiences (incl iOS) have been evaluated at some level within major destinations. I think there's value in exploring this option.

**Comment by feld** [comment]
Make sure it renders on Safari so Apple execs can read it

**Comment by starburst1977** [comment]
Totally agree with @auchenberg. Excellent write up. We need more browser choice on iOS. And I agree with @addyosmani that we perhaps should get in touch with the Safari Evangelists & Developers to make ourselves heard.

**Comment by therealmarv** [comment]
You start this petition page. This thread is already going kind of viral on HN ;) There are special services like https://change.org for get you started. No need to reinvent the wheel IMHO.

**Comment by 1vasari** [comment]
:+1: for https://change.org

**Comment by SarahX** [comment]
We should post this page to every tech link-sharing site (I saw this from Hacker News), so that's good. I would say post this to r/technology, but I think it's down right now due to issues between mods and admins. We need to make sure those in the web community understand the issues going on here. Unless we do this, we will never get enough people to actually make a difference.

**Comment by ileitch** [comment]
Provocative catchline idea for the site:

'Apple listens to Taylor Swift (https://twitter.com/cue/status/612824947342229504) and indie artists, but will they listen to developers?'

**Comment by CedricAlb** [comment]
+1 for change.org

**Comment by cibernox** [comment]
Another +1 for a global petition in change.org

**Comment by Shinta** [comment]
+1

**Comment by SarahX** [comment]
+1 for change.org.....it's a great way to get signatures.

**Comment by ngryman** [comment]
:+1:

**Comment by SarahX** [comment]
Who is creating the petition? just wondering :D

**Comment by auchenberg** [comment]
+1  for change.org

Is there a (native) english speaker who could draft up the petition text in a Gist? I'm about to catch a long flight, so I can have a look in 12-15 hours.

**Comment by SarahX** [comment]
Yeah, I am a native English speaker. I could create a draft. @stuartpb do you want to help contribute?

**Comment by SarahX** [comment]
Here it is: https://gist.github.com/SarahX/ec5525291686e2b83c5b

Fell free to contribute everyone....this is a community effort and everyone should input.

**Comment by egojab** [comment]
@SarahX the content blocking tools in iOS 9 aren't to push people to the app store; it is to improve the user experience on the mobile web (isn't that what you want?). Web ads are out of control. Kudos to Apple for bringing mobile Safari in line with desktop browsers with regards to giving users the ability to avoid the scourge of garbage ad practices in place on the web today. The "walled garden" vs. "open platform" comparison isn't relevant here. That rhetoric is tired, and inaccurate.

This is starting to read a lot more like it has little to do with getting Apple to improve Safari on iOS, and more to do with developer laziness. Developers who want to reach iOS users need to consider how iOS users use their devices.

**Comment by SarahX** [comment]
@egojab of course I want to prove the mobile web experience...and sadly web ads are sometimes very out of control. I 1000% agree that devs who want to reach iOS users need to consider how iOS users use their devices, as UX matters so much today. It' should not be about laziness, instead developers should work just as hard to meaningfully engage with iOS users the way it is now, but we should still encourage Apple to stay up to date with Safari. Thx for pointing that out, sometimes I'm wrong on topics, that's ok.

**Comment by RangerMauve** [comment]
@egojab Take into consideration that it's not just about features that are only available in the latest Chrome. Safari is lacking many features that are already implemented in the other major browser vendors (Firefox, ie11, Opera). As well Safari has implemented some features very poorly, for example IndexedDB.

**Comment by SarahX** [comment]
IndexedDB is not supported on Safari 7.1, but it does have partial support on more recent versions, although according to Can I Use on iOS8 it can sometimes be seriously buggy.

**Comment by egojab** [comment]
@RangerMauve I agree that Safari can certainly catch up in some of the more bleeding edge feature areas, but it's also important to understand that Safari is build for the web as it exists _today_, not for the web as it may be used in the future; they aren't making a browser for bleeding edge experimentation, they are building a browser for the vast majority of users and websites as they exist now, and doing so in a manner that seems focused on the user experience more than the developer experience. That may be frustrating for me as a developer, but ultimately, if it results in a better end-user experience, it's difficult to find much fault there.

I certainly do want Apple to be more engaged in developer and browser related discussions; What I don't want though, is for Apple to simply follow in Google's footsteps and bloat the browser with a whole bunch of "features" that have noticeably and measurably resulted in Chrome being an inefficient, bloated, beast, that is no longer best for users, even if it is "best" for developers.

**Comment by caffinatedmonkey** [comment]
:+1:

**Comment by RangerMauve** [comment]
@egojab I'm not sure why you're so fixated on Chrome when it's just one of the players out there. Features aren't being developed and implemented only in Chrome, it's pushed forwards by standards bodies and Firefox is also big on developing and deploying new features. Also the idea of Chrome becoming bloated is new to me, do you have any references that support that claim?

**Comment by RangerMauve** [comment]
Sorry, the main point I was trying to make with that is that Safari not implementing features isn't just about not following the path set out by Google, but it's related to them ignoring the web community as a whole when they don't participate in it.

**Comment by egojab** [comment]
@RangerMauve Only because the larger discussion I see surrounding Lawson's original article elsewhere; which seems to have started the whole feeding frenzy; is heavily skewed towards Chrome fans just wanting Safari to be more like Chrome, or just generally pushing for Apple to allow the full Chrome browser on iOS. 

As I said in my previous comment, I do want Apple to be more engaged and support more bleeding edge features, I just think the rhetoric surrounding this right now is more alarmist than is necessary or realistic. 

As for Chrome being inefficient and bloated, there are plenty of articles and stories about that, most widely as it relates to battery life on devices, but also in memory and CPU usage.

**Comment by RangerMauve** [comment]
I can agree that it might be a bit alarmist, but I disagree that it's just about Chrome fans wanting Safari to be more like Chrome, I think it's fans of modern browsers in general that are keeping up to date with new standards.

**Comment by egojab** [comment]
I don't think it's _just_ about Chrome fans wanting Safari to be more like Chrome - I just think that there is a lot of that underlying the conversation.

**Comment by SarahX** [comment]
I don't think Safari has to be more like Chrome...it can still adhere to Apple's minimalist design and UI standards yet still implement modern web standards. It's not just Chrome fans asking Apple to improve Safari...it's Firefox users and I probably think some of us could be typing this on our iOS devices as well. We are not looking for Safari to change to be more like Chrome, we want a better dev and user experience on the web in general.

**Comment by mryellow** [comment]
Nothing to do with Chrome fans vrs Safari fans. It's about not having to wait years for Apple to slowly breakdown, eventually acquiesces and allow software on their devices, to save their market share.

We can wait 5+ years of hamstrung innovation as they attempt to run the vendor locking strategy, one which has failed time and time again, or they can get with the program and remain relevant into the future while keeping developers on-side.

**Comment by stuartpb** [comment]
Hey all - I'm about halfway through catching up on this issue (had my hands full for a few hours with https://botbot.me/freenode/dokku/2015-07-03/?msg=43597775&page=3). I'm up to about https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-118372190 - I'll have something more to say after I've finished reading the (much more dense) further replies between then and now.

**Comment by stuartpb** [comment]
Okay, I'm caught up.

Yeah, in general the messaging I'm trying to go for here is more geared toward asking Apple to *let iOS users have a choice*, with regards to supporting the view expressed by @RangerMauve that browsers should move forward to embrace new standards (browsers should be liberal), versus supporting the view expressed by @egojab that web browsers should be reticent to adopt new standards (browsers should be conservative). Right now, Apple is imposing their conservative web views on *all* their customers, **and that's a problem**.

**Comment by stuartpb** [comment]
It's fine if Apple doesn't want to introduce new standards into Safari (even though the attitude I'm picking up isn't so much that they're *opposed* to integrating new standards as that they're *disinterested* in doing so), but the problem is in denying other vendors who do want to introduce these standards in *their* apps.

**Comment by SarahX** [comment]
So is the real issue not just about specific standards (un)compliance in Safari but the lack of browser choice on iOS? That makes sense but it is highly unlikely that Apple will try to add more competition by opening up to Chrome/Firefox/Opera/whatever.

**Comment by stuartpb** [comment]
Pinging Apple devs mentioned by @addyosmani (or best guess): @jondavis @kdecker @hober

**Comment by SarahX** [comment]
Once we get the petition up and running we should post it to HN and Reddit (if the tech subreddits are back up). We could potentially get thousands of views from those sites.

**Comment by stuartpb** [comment]
@SarahX The issue is specific standards (un)compliance in Safari *compounded with* the lack of browser choice on iOS. Fixing *either side* of this would make this less of an issue: fixing the latter is the one that would require significantly less effort from Apple, with less potential for friction going forward. (Ideally, it would be nice if Apple did both, but really they just need to let *somebody* fix it, which is what the latter fix is.)

**Comment by SarahX** [comment]
I agree...a hybrid mix of both would be the best option but I also feel that having more browser choice on iOS would eventually solve the first problem as they would need to be more competitive.

**Comment by stuartpb** [comment]
@SaraX Yes. See also https://kenneth.io/blog/2015/07/03/safari-isnt-the-problem-but-the-lack-of-browser-choice-in-ios-is/ as posted above by @addyosmani.

**Comment by SarahX** [comment]
@addyosmani makes a good point in the blog post where it says that these issues are holding back innovation on the web. The industry is usually a move fast(ish) while (sometimes) breaking things type of industry, and innovation is a driving force. Perhaps Apple is fearful of showing users too much outside their control?

**Comment by stuartpb** [comment]
Looks like somebody went ahead and started a change.org petition: https://www.change.org/p/apple-allow-true-other-browsers-on-ios

**Comment by stuartpb** [comment]
Will update the "Sign the petition" link momentarily (linking to this thread as "Discuss the problem").

**Comment by stuartpb** [comment]
Continuing from https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-117792929 and https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-118435712, I just sent this email:

> Date: Fri, 03 Jul 2015 17:43:51 -0700
> From: "Stuart P. Bentley" <stuart@testtrack4.com>
> To: team@fightforthefuture.org, info@eff.org, action@eff.org, 
> jond@ingenesis.net, kdecker@apple.com, hober0@gmail.com
> Subject: Opening the Web Platform for iOS
>
> Hi, I started http://www.safari-is-the-new-ie.com to raise awareness of 
> a serious (albeit somewhat technical) problem with the way Apple handles 
> browsers on iOS (further information on these linked pages):
>
>  * http://nolanlawson.com/2015/06/30/safari-is-the-new-ie/
>  * https://kenneth.io/blog/2015/07/03/safari-isnt-the-problem-but-the-lack-of-browser-choice-in-ios-is/
>  * https://github.com/h5bp/lazyweb-requests/issues/181
>
> Considering this is an open internet issue (specifically an open Web 
> issue), do you have any suggestions / thoughts / resources on what we 
> can do to make it better?

**Comment by SarahX** [comment]
If EFF and FFTF supports this they could make it one of their headlines and drive traffic to the petition site...that's another great way to get more people aware of what's going on bc so far this issue has a small follower base.

**Comment by SarahX** [comment]
What does true other browsers mean? This is fairly unclear.

**Comment by stuartpb** [comment]
Right now iOS has "false other browsers", in the form of apps for Chrome (and I *think* Firefox?) that are using WKWebView under the hood (ie. they're not *truly* other browsers).

**Comment by SarahX** [comment]
Good point....but some people might not understand what this means and it is in the title of the petition. I was just thinking it should be more clearer as this can be confusing to those who do not fully understand all of the technical aspects surronding it. Some might not sign it if they are confused at the first thing they read. Just a suggestion..the rest looks great.

**Comment by stuartpb** [comment]
You should tweet at https://twitter.com/t0mas, the petition's author.

**Comment by SarahX** [comment]
Thx but I regret to say I don't have Twitter. I just read other people's feeds...I'm too lazy to sign up. Does he have a GitHub?

**Comment by egojab** [comment]
@stuartpb I'd like to clarify that my stance is not "web browsers should be reticent to adopt new standards" It is that I think it's important to reach out to understand _why_ Apple has chosen to adopt the standards they have, and not yet adopted others. I would absolutely like to see more advanced, experimental features in Safari, but I don't think it's appropriate to take them to task over it without at least attempting to understand their choices first. That is why I fully support attempting to gain more involvement from Apple in discussions and events surrounding these features and issues.

Furthermore, I'd argue that you're not quite fit to take up the mantle of such a cause if you don't actually use the products, or don't have an understanding of what is or is not available on them. To say "and I _think_ Firefox" shows that you have a lack of knowledge on the subject, and seemingly a lack of desire to learn about it; as it would have only taken a couple of minutes of research to understand that there is, in fact, no Firefox browser on iOS.

Lastly, the whole tone of this discussion is going to have a poor chance of gaining any serious following when you keep saying things like "Apple is fearful of showing users too much outside their control". That is inaccurate and insulting to Apple, and to the users you're claiming to want to help. I think there is more hope if your attempts were not laced with the divisive "us vs. them" language creeping in here.

**Comment by stuartpb** [comment]
@egojab

> Lastly, the whole tone of this discussion is going to have a poor chance of gaining any serious following when you keep saying things like "Apple is fearful of showing users too much outside their control". That is inaccurate and insulting to Apple, and to the users you're claiming to want to help. I think there is more hope if your attempts were not laced with the divisive "us vs. them" language creeping in here.

You realize @SarahX and I are two different people, right? :smile:

**Comment by egojab** [comment]
@stuartpb Yes, I do, sorry, that was in response to @SarahX. I should have clarified.

**Comment by SarahX** [comment]
@stuartpb I did not anything about "I think Firefox". And yes, we are different people. I do also believe that we should reach out to Apple and have discussions on how to improve the web experience for both users and developers. I am not trying to stir up hate for Apple, I too, use some of their products and enjoy using them as well.

**Comment by stuartpb** [comment]
@egojab 

> Furthermore, I'd argue that you're not quite fit to take up the mantle of such a cause if you don't actually use the products, or don't have an understanding of what is or is not available on them. To say "and I think Firefox" shows that you have a lack of knowledge on the subject, and seemingly a lack of desire to learn about it; as it would have only taken a couple of minutes of research to understand that there is, in fact, no Firefox browser on iOS.

I used "I *think* Firefox" as a shorthand for "other browser apps", since I *am* very busy trying to moderate this issue, **and** [draft a standard for static file server routing](https://github.com/isaacs/github/issues/408), **and** [work with Dokku maintainers who expect to be "releasing a 0.4.0 tomorrow and then a 0.5.0 Sunday" to integrate projects that have existed separately for almost a year](https://github.com/progrium/dokku/pull/1228#issuecomment-118424269). I apologize that I don't have time to look up *every last pedantic barely-relevant detail* when answering individual requests for clarification.

**Comment by stuartpb** [comment]
Also, this is why I'm leaving the majority of the drafting/planning/outreach on this issue to users like @nolanlawson, @auchenberg, tomas0 on Twitter, and @addyosmani, handling bits and bobs of pinging and pages here and there.

**Comment by SarahX** [comment]
@staurtpb: Should I post to HN now?

**Comment by egojab** [comment]
I'm sorry if you're offended by the criticism. I think it's important to actually understand the system you're wanting to fix. Especially when doing so would not actually take a tremendous amount of time. If you're just going to blindly follow some alarmist blog post that went viral online, fine, but then maybe don't open it up for discussion on a public forum.

I'll respectfully unsubscribe from here, as this feels more like a pitchforks and torches posse than a discussion of what's actually best for web users and web advancement.

**Comment by stuartpb** [comment]
@SarahX t0mas [does not appear to have a GitHub account](https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-118436764) (click through that org link, then to the repo, and you'll see that his contributions are attributed without linking.)

@egojab :+1:

**Comment by SarahX** [comment]
@stuartpb thx for the info....it's not a huge problem to deal with. Perhaps someone else along the road will fix it eventually. As long as it does not distract from the main purpose it's okay.

**Comment by stuartpb** [comment]
Yeah, I'm working on inviting him to join this issue now (once I can stop writing "offended" responses to concern trolls saying I'm "blindly follow[ing] some alarmist blog post that went viral online" without "actually understand[ing] the system you're wanting to fix").

**Comment by stuartpb** [comment]
[tweeted](https://twitter.com/stuartpb/status/617145454002745344)

**Comment by SarahX** [comment]
@staurtpb thanks that will be a major help to hear his input as he started the petition.

**Comment by andysolomon** [comment]
Am I allowed to totally disagree with this Petition.

**Comment by t0mas** [comment]
Hi all,

I do have a Github account :-) same username as Twitter but I see my profile doesn't have a last name which is probably why you couldn't find it.

Looks like there is a suggestion here for a better text, so I'll take a look at that and update the petition. It will take me a bit of time, since I'm currently abroad and about to hop on a flight back home. Feel free to provide suggestions in the mean time.

**Comment by Garbee** [comment]
@andysolomon It's all good to dislike it (and voice that here.)

Honestly, this moved way too quickly. It went from an idea to petition in a matter of a few days. Even further, as far as I can tell no attempts to reach out to Apple first and see if they would be willing to work to address developers concerns.

Everyone should calm down a bit, put the petition on hold, and we should try to reach out to people at Apple to try and get some official word on things before going any further. We don't need a pitchfork-mob mentality which this seams to have accumulated.

**Comment by andysolomon** [comment]
@Garbee I agree. I understand everyone's concerns about Safari and mobile Safari. I think Apple is attempting to move forward into a new paradigm shift of users not being reliant on a browser and URLs. Let's remember how old the web is, and how archaic most web technology is.  HTML was created in 1993. Do we really expect to be using HTML and CSS in 2023? I think you can expect Apple( and others will follow ) to move towards native software that truly empowers the users, and enhance their user experience. 

Yes, a web browser is native software, but it promotes more webapps and web software. I think the 🍏WATCH is the first step towards this new paradigm shift of catering to the user and giving them exactly what they want. Obviously this is an unpopular opinion here, but lets seriously consider the platforms we will be developing for in 2020. Most proponents of the "safari-is-the-new-ie" fame were born in the 20th century, maybe the web as we know it is not equipped for the 21st century.

**Comment by SarahX** [comment]
@andysolomon Yes...times are changing. I was born in the 21st Century, one of those so-called "digital natives", and the last birth year of the millenials. And no, we won't be using HTML and CSS in 2023, which is totally OK. Technology gets old at a very quick pace. I do think we should reach out to Apple and maybe get a discussion going about Safari. But if they won't listen, we should act as the time to fix Safari is now, not in 5 years when no one is going to care anyways.

**Comment by auchenberg** [comment]
Okay, so I now I had a chance to catch-up with this thread, and I think we have started to mix-up the multiple agendas that are present in here.

@SarahX I just had a look at your petition text, and your text is about improving Safari as a browser. 

@t0mas's Your petition that's about the lack of browser choice in iOS

These two objectives are quite different, but I don't the rule each other out, they are just objectives each with their own goals, and therefore I see a need to separate them into different petitions.

As I see it, these are the potential actions of this thread:
### Actions

1. Build a website for safari-is-the-new-ie.com, with the overall objective to get Apple more involved with web standards, get more frequently updates of Safari and get Apple to have a strong presence in web community

2. Start a petition about 1) with the objective to collect signatures to prove the demand for a stronger web presence by Apple

3. Start a petition about the wish for a free browser choice in iOS with the objective to collect signatures to prove the demand. My blog-post could be used as the "informative website", as there's already plenty of comments on Hackernews, Twitter, and the blog itself.

I suggest we can use my wording for the browser choice petition:
```
Dear Apple

We are a bunch of passionate people that really want to make the web better for everyone, but you are holding us back.

It’s all right that you don't have the resources to drive the web (or that you simply don't want to any more), but now is the time to open up the browser lock-in in iOS and allow us to innovate.

The undersigned.
```

If necessary I suggest we fork out this discussion into two threads, if that helps keeping the conversation(s) focused.

**Comment by SarahX** [comment]
@auchenberg sorry that my draft was incomplete...I think that a hybrid approach is even better. More importantly, the developer community should have a discussion with Apple and see what they think and what they can do to solve the problems. We do not need to fork the conversation unless someone really wants to as we should hear as much input from anyone as we can right now. Both problems feed off of each other, lack of browser choice means that they don't have to improve Safari.

**Comment by andysolomon** [comment]
If you hope for Apple to take any of this seriously you may want to reference Steve Jobs' [Thoughts On Flash]( https://www.apple.com/hotnews/thoughts-on-flash/ ). 

I believe this is what @nolanlawson is proposing, a very detailed letter describing why its so important to have alternate browser engines running on iOS. You don't need some designed website with a clickbait URL, and you may not even need a petition. Just a well written letter showing how much you care, and how you plan to change the world and solve real world problems by having browser engine options on iOS.

@addyosmani proposed speaking with a devoted Safari or iOS engineer. You may want to get someone like that to deliver the letter.

Like I've mentioned before, I don't think we need an alternate browser; Simply because the users are not asking for another browser. Users want ways in which they could interact with their phones less.  Apple believes in the open web, and this is something that I'm sure Apple has considered several times in the past, but the approach must be very **empathetic and scientific**, if you hope for Apple to entirely change its business goals.

**Comment by SarahX** [comment]
@andysolomon Should we all sign the letter? Be "we all" I mean anyone who is concerned about these issues. Or should it just be an open letter? How will Apple know that this concerns more than just a few people? Just wondering

**Comment by andysolomon** [comment]
@SarahX I think Step One will be to write the letter. Step Two will be to show the letter to the web community and get signatures. Step Three would be electing someone to deliver the letter.

As of today I would not sign the petition, because there is no clear plan for action. The letter should be good enough to where anyone(webDevs, iOSDevs, etc...) would agree. Otherwise it just looks like a tangent on the web. Again reference Steve Jobs.

**Comment by SarahX** [comment]
@andysolomon Yes, if we are able to get iOS devs and even those working at Apple to support us then it would seem much more reasonable and they would be kind of forced to listen to us. If we are able to prove that these issues are leading to bad UX as well then we have a high chance of them listening to us.

**Comment by cibernox** [comment]
I think that *if one concern* has to be more present than the other in the petition has to be to open the market to have other browsers with their own engines in iOS.

Not only because in my opinion **this is the root cause** of the problem (if apple had some competence in their ecosystem they would have to respond to user expectations, because others do) but also for the moral implication of restricting user's freedom to choice the browser they prefer with no real benefit in security or anything else for nobody.

In other words, it's ok that apple is not interested as other companies in improving they flagship browser, but if that's the case, please, let those other companies offer they software to the users.

**Comment by auchenberg** [comment]
@SarahX I disagree. I think a combined petition of everything proposed in this thread (safari improvements, more rapid release cycle, better developer evangelism, and a free browser choice), is simply too much and too unfocused. As it stands right now I won't be able to sign such combined petition, as I simply don't agree all of the above. I think many people with have it the same way, as there's simply too many variables people can agree/disagree on, and this will hinder people signing the petition. 

If our goal is to get as many people as possible to support this petition, the messaging should be crystal clear. I imagine a petition similar to the [Extensible Web Manifesto](https://extensiblewebmanifesto.org/), and it will be "signed" by the people who support the petition.

@andysolomon When reading @nolanlawson's [comment here](https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-117708409) in this thread, and his blog post it's quite clear that the post isn't about the lack of free browser choice in iOS, but how Safari is lacking behind, and how he feels Apple in is slacking it's web efforts. 

@cibernox Agreed, and that's what I'd argue for a petition focused about the lack of browser choice.

**Comment by jsemu** [comment]
Apple left in a bunch of regressions within mobile safari when they updated iOS from 7 to 8. A bunch of webpages crash out the web process and get into a reload loop with a generic error message.

Namely http://jsemu.github.io/gba/ used to run on an iphone 5S running iOS 7, now it often reload loops on iOS 8. Unpredictable too on when safari decides to.

**Comment by SarahX** [comment]
@auchenberg So are we going for a more specific approach? This is fine but we all need something to agree on. Should the lack of browser choice be the main issue?

**Comment by stuartpb** [comment]
Just because it looks like people missed it, [the email I sent earlier](https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-118446712) was sent to the three Apple developers @addyosmani [suggested reaching out to above](https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-118326076), as well as FFTF and EFF.

I'm looking to reach out to the devs/reps inside Apple *at the same time* as the general public, because, from my (first-hand and second-hand) experience with working inside a big company, it's just as hard for a few insiders to push for changes as it is for a few outsiders, without support from the other side. It's my firm belief that we need both internal *and* external support for Apple to change here.

**Comment by sghall** [comment]
Just wanted to add my voice.  In line with @auchenberg and @cibernox's comments.  Apple should decide if they want to work harder on Safari and what features they want to focus on.  The petition should be about browser choice.  If users have options, in the long run, things will work themselves out.  That said, the Safari issues are serious and, left unresolved, will slow innovation in the medium term.  It would be nice if Apple would be a better steward of the open web, but that has got to be their choice.

**Comment by jeroenhabets** [comment]
Apple's Safari team not being actively involved in the web community has lead to Safari falling behind, hindering the web community to provide the optimal user experience.

If I read correctly, most commenters in this thread believe that optimal user experience on iOS requires opening up the iOS platform to alternative browsers and thereby introducing competition. Assuming alternative entities will invest in their browsers on iOS, which is a rather safe assumption, and thereby providing this improved user experience, this would put pressure on Apple to collaborate with the web communitiy and invest in Safari again.

I would gladly sign a petition requesting Apple to allow alternative browsers to iOS. However, I would also support an alternative petition merely requesting Apple to become actively engaged in the web community again, and invest in Safari to fix the issues (bugs and missing features).
Though, I am a proponent of open platforms and would not want to abandon a petition for opening up iOS, I could imagine the second petition to have a higher chance of success as it is more pragmatic.

**Comment by albertorestifo** [comment]
:+1:

**Comment by SarahX** [comment]
I think it is important to push for browser choice as competition leads to better products. If Chrome/Firefox/Opera have better UX (for users and devs) then Apple will be forced to fix Safari or lose part of their user base. However, without actually opening up talks with Apple, we might face similar problems in the future. Otherwise, it sounds great. Let's start getting more support.

**Comment by jpsc** [comment]
Apple like any other company will do whatever benefits their business model. They will not do anything without a reason business decision behind. If they make the users/developers happier is because it interests them.

**Comment by cibernox** [comment]
@jpsc This petition could at least let them know how unhappy we feel because of them.

**Comment by jpsc** [comment]
@cibernox I know. I'm not saying we should not do anything. We should make our voices be heard. On a different topic, when was the last time a online petition of anything ever made a difference :santa: ?

**Comment by cibernox** [comment]
@jpsc Around one week ago, when an open letter of Taylor Swift made Apple change the conditions of Apple Music

**Comment by jpsc** [comment]
@cibernox I am a conspiracy guy. I think that it was all cooked :walking: But I was talking more about a pettion on its core idea; a bunch of people wanting something and getting it. That was a pop star that had already said no to spotify before.

**Comment by stuartpb** [comment]
I'm hearing increasing amounts of conspiracy rumbling in here, along with increasing rumbling over people behaving from a "pitchforks and torches" mentality.

I think we should take care to [assume good faith](https://en.wikipedia.org/wiki/Wikipedia:Assume_good_faith) here (both regarding Apple, and in interacting with each other) and follow [Hanlon's razor](https://en.wikipedia.org/wiki/Hanlon%27s_razor):

> Let us not attribute to malice and cruelty what may be referred to less criminal motives. Do we not often afflict others undesignedly, and, from mere carelessness, neglect to relieve distress?

**Comment by andysolomon** [comment]
We should setup a conference call( google hangout ), and record it. Give everyone a chance to voice their opinions, and leave the call with something to act on.

**Comment by stuartpb** [comment]
@andysolomon

> We should setup a conference call( google hangout ), and record it. Give everyone a chance to voice their opinions, and leave the call with something to act on.

Maybe, soon - I want to give the parties I emailed earlier some time to reach back before we start scheduling any meetings. (Since I sent that email at P5:45 on the Friday opening a three-day weekend, I'm planning on following up Tuesday or Wednesday in case they missed it.)

**Comment by auchenberg** [comment]
Okay, pitching in here again to get the ball rolling:

There's absolutely no need for the "pitchforks and torches" mentality here. The goal of a given petition is to collect interest and simply to show Apple that a larger group of people would like to see Apple move in this direction. 

We should focus on getting the petition text going, instead of scheduling meetings, reaching our to Apple contacts, etc. There's currently no **consensus** on the message we want communicated to Apple, so let's focus the energy on get the message sorted out, before thinking in PR, Tweets, etc.

I'm currently on a terrible connection, but my suggestion text from https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-118523821 still stands to the browser-choice petition.

It's there's no better alternatives by tomorrow I suggest we move forward with the text, and get @t0mas to update the existing petition on change.org?

**Comment by t0mas** [comment]
Sure, although we may want to extend your text a bit to explain to people less familiar with the situation what the current situation actually is. Especially because on the surface it may seem that there are three browsers for iOS (Safari, Chrome, Firefox) while underneath Apple forces all to use the Safari engine. I'm assuming many people don't know about that and for those people it may seem like we're asking Apple to do something they have already done.

**Comment by auchenberg** [comment]
@t0mas Really good point. Do you or should I take a stab at extending the message? I can write something up this evening, if my jet-lag allows me to stay awake.

**Comment by t0mas** [comment]
If you have the time a suggestion would be very welcome, bit swamped at work at the moment.

**Comment by scshepard** [comment]
A couple of thoughts: First, I'm not a Mac or ios guy. I do have several tech mac friends, and their response to your concerns: we don't care.So, does it make sense to compile a list of sites that don't work on Safari ios, or sites that require too much "workarounds". That might more traction with getting folks concerned. 
Also, I'm far more concerned about the appearance that apple via WebKit is setting with the slow updates, focus on non-standard solutions (lots of posts regarding the new touch and gesture javascript stuff), as well as the falling back on the "dizzy motion" setting health problem. I have lots of posts and tweets if anyone would like to get specifics.
Kinda reminds of the days when AOL first came on the screen. Tried many times to convince my friends that "AOL is NOT the internet." But once they tasted freedom, they never went back. Anyone care to move back to the days of corporate portals? I thought not.
The primary reaction from my very small sample: Safari works great, we have three browsers to choose from, don't bother me.
I'm enjoying the "lesson" from all the viewpoints.Steve

Date: Mon, 6 Jul 2015 02:02:37 -0700

**Comment by stuartpb** [comment]
In case anybody missed it, @nolanlawson posted a great follow-up to his original post yesterday: http://nolanlawson.com/2015/07/05/safari-is-the-new-ie-2-revenge-of-the-linkbait/

**Comment by talamaska** [comment]
Hi to all I like to see so much people having same problems as me with Safari. I think there were very constructive suggestions in the discussion till now. I also wouldlike to add my 2 cents onthis matter. First of all, did any of you watched the last wwdc webkit keynotes? Apple announced some good features on the webkit, but none of them was from the official w3c standards and didn't really care about them,cause they were not so critical on my opinion as having somestandards implemented like UserMedia, IndexedDb, full File Api or any of the others already implemented in Firefox or Chrome standards. What really frustrates me is that often Apple don'treally care about standards, this is why they have become new Ie. Remember that early 90s antitrust process vs MS, well it was not the reason for IE to become better on supporting w3c, it was the competition with other browsers/engines who were having a lot more user base. So basically what I'm saying is that the first step should be allowing other engines in iOS, and webviews. You can't enforce Apple to implement features on Safari, but you can make them open the iOS for competitors. The fact that I just learned the Chrome App is not the real Chrome is really frustrating, as Google didn't say anything about it.

**Comment by metrofun** [comment]
Also if iOS will become open to other browser engines then it will give a boost to hybrid app development with CrossWalk for example (e.i. kinda PhoneGap bundled with Chromium).

**Comment by simevidas** [comment]
Have you seen this feedback channel for Safari:  https://www.apple.com/feedback/safari.html

Maybe we could force Apple to make a statement if we’d get a lot of people to submit a predefined message (similar to those public campaign sites that were created to instruct people to “spam” the FCC). If Apple is a respectable company, they would at least have to acknowledge the feedback, which would be a small step in the right direction.

**Comment by stuartpb** [comment]
Okay, I never heard back from any of the people I sent my email to (I re-sent it to the EFF [after a spur-of-the-moment exchange on Twitter](https://twitter.com/amullionaire/status/628688110293921792)), but I'm back in the mindset for this.

I also took a moment to look back at some of the reaction from this from bck in July (when I was too busy throwing this page together to look at what the widespread reaction to it was), and it has been aggravating - [particularly from Safari devs themselves](https://twitter.com/grorgwork), who think developers having to bend over backwards to fix problems on their behalf is a *joke*. (I really don't think they realize how bad it is - all their deflections amount to stuff like "Sorry we're not natively supporting stuff only one browser supports! Guess you'll have to wait a few more years for Custom Elements without a polyfill!", while handling criticisms of their IndexedDB like "huh, I guess, whatever".)

So, here's what I see from safari-is-the-new-ie.com going forward: I want to have a ~paragraph or so explanation of how exactly the problem is occurring, culturally, **in a way accessible to the average user**, along with some synopses of **each major issue** users will only encounter on Safari due to it being a pain to support (so developers can send users who email them about their webcam app not working on iOS to this page saying "sorry, it's Apple's fault").

What I'm looking for now is some people who can submit content and pull requests to add that - right now, I'm running late for an event, so I'm going to have to cut this comment here.

**Comment by andyps** [comment]
Why is your petition closed on change.org now?

**Comment by stuartpb** [comment]
I don't know - reach out to whoever started the petition (see https://github.com/h5bp/lazyweb-requests/issues/181#issuecomment-118436764).

**Comment by styfle** [comment]
Webkit is finally [starting development](https://bugs.webkit.org/show_bug.cgi?id=174541) on service workers. Has iOS opened up 3rd party browser support? Maybe in iOS 11?

**Comment by simevidas** [comment]
No.
